JASP-Engine/JASP/R/principalcomponentanalysis.R

```diff
@@ -15,850 +15,430 @@
 # along with this program.	If not, see <http://www.gnu.org/licenses/>.
 #
 
-
-PrincipalComponentAnalysis <- function(dataset = NULL, options, perform = "run",
-																			callback = function(...) list(status="ok"), state = NULL, ...) { 
-	return(mainFunctionPCAEFA(type = "pca", dataset = NULL, options, perform = "run",
-																			callback = function(...) list(status="ok"), state = NULL, ...))
+PrincipalComponentAnalysis <- function(jaspResults, dataset, options, ...) {
+  jaspResults$addCitation("Revelle, W. (2018) psych: Procedures for Personality and Psychological Research, Northwestern University, Evanston, Illinois, USA, https://CRAN.R-project.org/package=psych Version = 1.8.12.")
+  
+  # Read dataset
+  dataset <- .pcaReadData(dataset, options)
+  ready   <- .pcaCheckErrors(dataset, options)
```

Comment:
Nitpick: it is quite tempting to think the `ready` boolean is the outcome of the error check. This is not the case (it has nothing to do with what else happens in that function).. but to find out you have to look in the error check. Maybe do this separately?

Reply:
fixed.